furusiya/function_mess.py

```diff
@@ -461,8 +467,9 @@ def handle_keys():
             if user_input.text == 'g':
                 # pick up an item
                 for obj in Game.objects:  # look for an item in the player's tile
-                    if obj.x == Game.player.x and obj.y == Game.player.y and obj.item:
-                        obj.item.pick_up()
+                    obj_item = obj.get_component(Item)
```

Comment:
This should be replaced with `has_component` and should be called `is_item` for clarity